mem0/vector_stores/pgvector.py

```diff
@@ -0,0 +1,257 @@
+import json
+from typing import Optional, List, Dict, Any
+from pydantic import BaseModel
+
+try:
+    import psycopg2
+    from psycopg2.extras import execute_values
+except ImportError:
+    raise ImportError("PGVector requires extra dependencies. Install with `pip install psycopg2`") from None
+
+
+from mem0.vector_stores.base import VectorStoreBase
+
+class OutputData(BaseModel):
+    id: Optional[str]
+    score: Optional[float]
+    payload: Optional[dict]
+
+
+class PGVector(VectorStoreBase):
+    def __init__(
+        self,
+        dbname,
+        collection_name,
+        embedding_model_dims,
+        user,
+        password,
+        host,
+        port
+    ):
+        """
+        Initialize the PGVector database.
+
+        Args:
+            dbname (str): Database name
+            collection_name (str): Collection name
+            embedding_model_dims (int): Dimension of the embedding vector
+            user (str): Database user
+            password (str): Database password
+            host (str, optional): Database host
+            port (int, optional): Database port
+        """
+        self.conn = psycopg2.connect(
+            dbname=dbname,
+            user=user,
+            password=password,
+            host=host,
+            port=port
+        )
+        self.cur = self.conn.cursor()
+
+        self.create_col(collection_name, embedding_model_dims)
+    
+    def create_col(self, name, embedding_model_dims):
+        """
+        Create a new collection (table in PostgreSQL).
+
+        Args:
+            name (str): Name of the collection.
+            embedding_model_dims (int, optional): Dimension of the embedding vector.
+        """
+        self.cur.execute(f"""
+            CREATE TABLE IF NOT EXISTS {name} (
+                id UUID PRIMARY KEY,
+                vector vector({embedding_model_dims}),
+                payload JSONB
+            );
+        """)
+        self.conn.commit()
+
+    def insert(self, name, vectors, payloads = None, ids = None):
+        """
+        Insert vectors into a collection.
+
+        Args:
+            name (str): Name of the collection.
+            vectors (List[List[float]]): List of vectors to insert.
+            payloads (List[Dict], optional): List of payloads corresponding to vectors.
+            ids (List[str], optional): List of IDs corresponding to vectors.
+        """
+        if payloads is None:
+            payloads = [{}] * len(vectors)
```

Comment:
This feels weird. Why do we need to do this?

Reply:
Ohh wait this was not supposed to be there.

I was basically checking that if payload is none we would have to insert some value along with vectors into the db, because when I was testing somewhere it caused such error.

But then it was supposed to removed because in `insert` we won't have a case where payloads or ids would be None.